### PR TITLE
Record interaction between GUI and LS

### LINKED ISSUE #8328: Record interaction between GUI and LS
As a step towards [chaos testing](https://github.com/enso-org/enso/issues/8247) we would like to record all messages that are being exchanged GUI and LS so that they can be inspected. Let's make the option parameterized and off by default. Should be able to specify the location where messages are being recorded, with some reasonable default.

The work is also a step towards potentially automatically replaying messages being sent by GUI.

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #8328

When the profiling is enabled, the language server messages are also recorded.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by Akirathan — COMMENTED** [review]
Please, don't forget to add docs in some reasonable place about what will be the default file name, etc.

File: engine/language-server/src/main/scala/org/enso/languageserver/boot/ProfilingConfig.scala [hunk 1]
```diff
@@ -23,19 +23,68 @@ case class ProfilingConfig(
...
+      )
+    )
 }
+
 object ProfilingConfig {
 
+  private val EventsLogSuffix    = ""
   private val EventsLogExtension = "log"
+
+  private val MessagesSuffix          = "-messages"
```
Comment (JaroslavTulach) [inline engine/language-server/src/main/scala/org/enso/languageserver/boot/ProfilingConfig.scala:60]: How does one trigger this profiling on? Just `--profiling-path`?

File: engine/language-server/src/main/scala/org/enso/languageserver/boot/ProfilingConfig.scala [hunk 2]
```diff
@@ -23,19 +23,68 @@ case class ProfilingConfig(
...
+  private val TextMessagesExtension   = "txt"
+  private val BinaryMessagesExtension = "bin"
```
Comment (JaroslavTulach) [inline engine/language-server/src/main/scala/org/enso/languageserver/boot/ProfilingConfig.scala:62]: Does it make sense to split textual and binary messages? How are we going to replay them and keep order?

**Review by JaroslavTulach — CHANGES_REQUESTED** [review]
I don't have a clear idea how is this functionality going to be used. I'd like to see _"how-to record & replay(?)"_ document first.

**Comment by hubertp** [comment]
> I don't have a clear idea how is this functionality going to be used. I'd like to see _"how-to record & replay(?)"_ document first.

I don't think the _replay_ part is planned any time soon. It's going to be very hard to try to exactly replicate the state of the Engine at the point the request is being made.

**Comment by JaroslavTulach** [comment]
> > I don't have a clear idea how is this functionality going to be used. I'd like to see _"how-to record & replay(?)"_ document first.
> 
> I don't think the _replay_ part is planned any time soon. It's going to be very hard to try to exactly replicate the state of the Engine at the point the request is being made.

How do we want to verify the code in this PR works correctly then? A unit test?

> Unit tests have been written where possible.

I don't see any test. Neither description how to manually verify things are working OK.

**Pull request closed without merging**